source/docs/software/commandbased/commands-v3/what-is-command-based-v3.rst

```diff
@@ -0,0 +1,241 @@
+# What Is Commands v3?
+
+Commands v3 is WPILib's next-generation command-based framework for Java. It introduces imperative-style command writing using coroutines, allowing you to write robot behaviors as straightforward sequential code instead of complex chains of lambda expressions and decorators.
+
+.. warning::
+   Commands v3 requires **Java 21 or later** and is **Java-only**. If you need C++, Python, or earlier Java versions, use :ref:`docs/software/commandbased/commands-v2/index:Commands v2 Programming`.
+
+## Command-Based Programming Philosophy
+
+Like Commands v2, Commands v3 is a :term:`design pattern` for organizing robot software. It's not the only way to write a robot program, but it's effective for creating clean, maintainable, and reusable code.
+
+The key difference: **v3 allows imperative control flow** in addition to v2's declarative composition style.
+
+## Declarative vs Imperative: An Example
+
+Consider a command that extends a piston when a condition becomes true.
+
+**Declarative (v2 style, still works in v3):**
+
+.. code-block:: java
+
+   import org.wpilib.commands3.Trigger;
+
+   new Trigger(() -> condition.get())
+     .onTrue(pneumatics.run(coro -> piston.set(Value.kForward)).named("Extend"));
+
+**Imperative (v3 coroutine style):**
+
+.. code-block:: java
+
+   Command extendWhenReady = Command.noRequirements()
+     .executing(coroutine -> {
+       while (!condition.get()) {
+         coroutine.yield(); // Wait for condition
+       }
+       piston.set(Value.kForward);
+     })
+     .named("Extend When Ready");
+
+**Without command-based (manual state tracking):**
+
+.. code-block:: java
+
+   // In periodic():
+   if (condition.get()) {
+     if (!pressed) {
+       piston.set(Value.kForward);
+       pressed = true;
+     }
+   } else {
+     pressed = false;
+   }
+
+The declarative v3 trigger style is concise for simple actions. The imperative v3 coroutine style shines when you have complex sequential logic, loops, or conditionals.
```

Comment:
I don't think it's necessary to qualify these as the "v3" trigger style or "v3" coroutine style; just "trigger style" and "coroutine style" is enough. Readers already know we're talking about the v3 API